cmd/nodecmd/create_aws.go

```diff
@@ -152,7 +156,7 @@ func createEC2Instances(rootBody *hclwrite.Body,
 	keyPairName,
 	securityGroupName string,
 ) ([]string, []string, string, string, error) {
-	if err := terraformaws.SetCloudCredentials(rootBody, region); err != nil {
+	if err := terraformaws.SetCloudCredentials(rootBody, awsProfile, region); err != nil {
```

Comment:
can the parent of this also receive awsProfile as param? just try to minimize global var usage for flags